src/lib/utils/theme/useTheme.tsx

```diff
@@ -0,0 +1,27 @@
+import { createTheme, StyledEngineProvider, ThemeOptions } from '@mui/material'
+import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles'
+import { merge } from 'lodash-es'
+import React, { useMemo } from 'react'
+import defaultMuiTheme from './DefaultTheme'
+
```

Comment:
I got rid of the `SynapseTheme`. To override the theme, just use the MUI ThemeOptions type. We'll merge it with a default set of ThemeOptions